### PR TITLE
Updates for Scala Native 0.5.1

### PR DESCRIPTION
**Pull request by ekrich**
- [x] Waiting on https://github.com/typelevel/paiges/issues/615
- [x] PR is progress - https://github.com/typelevel/paiges/pull/617
- [x] Paiges is waiting for cats - https://github.com/typelevel/cats/issues/4587
- [x] Waiting for Paiges release - changed from draft so it can be reviewed.

### CODE REVIEW

**Comment by ekrich** [comment]
Do we want to update Scala.js to the latest?

**Comment by tgodzik** [comment]
> Do we want to update Scala.js to the latest?

I think that's fine, though probably in a separate PR?

**Comment by ekrich** [comment]
Oh, I was trying to update everything like `sbt-ci-release` and other things. I can back off and just touch what needs to be updated or all at once which I think might be better.

**Comment by tgodzik** [comment]
> Oh, I was trying to update everything like `sbt-ci-release` and other things. I can back off and just touch what needs to be updated or all at once which I think might be better. `sbt-native-packager` is a weird one because of release problems so not sure on that one.

it would probably be better one by one, but I wonder why it didn't happen via steward

**Comment by ekrich** [comment]
Ok, I will just back off to do only the required upgrades.

**Comment by tgodzik** [comment]
Ok, added metaconfig to the repo, that should help in this case. We can wait for the updates to show up, I just run the workflow

**Comment by ekrich** [comment]
That is great thanks - those can all go through as I wait.

**Comment by ekrich** [comment]
@tgodzik I will go ahead and rebase when all the updates get merged.

File: build.sbt [hunk 1]
```diff
@@ -122,8 +122,8 @@ lazy val core = crossProject(JVMPlatform, JSPlatform, NativePlatform)
     mimaSettings,
     moduleName := "metaconfig-core",
     libraryDependencies ++= List(
-      "org.typelevel" %%% "paiges-core" % "0.4.3",
-      "org.scala-lang.modules" %%% "scala-collection-compat" % "2.11.0"
+      "org.typelevel" %%% "paiges-core" % "0.4.3-35-4d6217a-SNAPSHOT",
```
Comment (tgodzik) [inline build.sbt:125]: Looks like we can switch to 0.4.4 ?

**Review by tgodzik — CHANGES_REQUESTED** [review]
Awesome! It's finally passing :tada: 

We can just update paiges to latest version and merge.

File: build.sbt (hunk 1 above)
Comment (ekrich) [reply build.sbt]: Updated waiting for CI to pass.

**Comment by tgodzik** [comment]
Should we do a new release?

**Comment by ekrich** [comment]
I wanted to update this in case someone ever gets back to trying to port the Scala Native `scalafmt`. We have multi-threading now so there could be less changes. The diff(ing) part was asked to be a separate project which seemed to be the biggest part of the angst of the large change.

**Comment by tgodzik** [comment]
Och yeah, I did release it separate because of that. I need to return to the PR :|

**Comment by ekrich** [comment]
> Och yeah, I did release it separate because of that.

Do you have a link - we might be able to use it in Scala Native for our tools which we are trying to port to Scala Native as part of a potential "native" tool chain.

**Comment by tgodzik** [comment]
We release is alognside munit https://mvnrepository.com/artifact/org.scalameta/munit-diff

**Pull request merged**